### PR TITLE
Bundle CA certs on linux if we have a single cert.pem file

### LINKED ISSUE #13399: Many automated builds/tests are broken with a libgit2 ECERTIFICATE error
The errors are along the lines of:

```
ERROR: LoadError: GitError(Code:ECERTIFICATE, Class:SSL, The SSL certificate is invalid)
 [inlined code] from libgit2/error.jl:96
 in clone at libgit2/repository.jl:95
 in clone at libgit2.jl:303
 in anonymous at /home/ubuntu/buildbot/slave/coverage_ubuntu14_04-x64/build/julia-1ed51ea4cd/share/julia/test/libgit2.jl:72
 in temp_dir at /home/ubuntu/buildbot/slave/coverage_ubuntu14_04-x64/build/julia-1ed51ea4cd/share/julia/test/libgit2.jl:62
 in temp_dir at /home/ubuntu/buildbot/slave/coverage_ubuntu14_04-x64/build/julia-1ed51ea4cd/share/julia/test/libgit2.jl:5
```

@staticfloat first pointed this out on a [buildbot](http://buildbot.e.ip.saba.us:8010/builders/coverage_ubuntu14.04-x64/builds/1119/steps/Run%20non-inlined%20tests/logs/stdio), but it's also happening on Travis:
https://travis-ci.org/JuliaStats/NullableArrays.jl/jobs/83049793
https://travis-ci.org/JuliaLang/JLD.jl/jobs/83031348
https://travis-ci.org/JuliaStats/PDMats.jl/jobs/83032826

### PR DESCRIPTION
**Pull request by staticfloat**
This closes #13399

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -448,6 +448,10 @@ ifneq ($(OS), WINNT)
 endif
 ifeq ($(OS), Linux)
 	-$(JULIAHOME)/contrib/fixup-libstdc++.sh $(DESTDIR)$(private_libdir)
+	# We need to bundle ca certs on linux now that we're using libgit2 with ssl
+ifneq ($(shell cat $(shell openssl version -d | cut -d '"' -f 2)/cert.pem),)
+	-cp $(shell openssl version -d | cut -d '"' -f 2)/cert.pem $(DESTDIR)$(datarootdir)/julia/
+endif
```
Comment (wildart) [inline Makefile]: So you decided on a file which is probably covered by GPL.

cc @tkelman

Comment (tkelman) [inline Makefile]: can you explain? thought this was from mozilla ca-certificates? though yes this should be mentioned in LICENSE.md, along with openssl while we're at it

Comment (wildart) [inline Makefile]: `ca-certificates` package on Ubuntu is covered by GPL 2

Comment (Keno) [inline Makefile]: Definitely MPL 2.0. The FSF has always maintained that compilers etc. do not affect the license of the code being transformed.

Comment (tkelman) [inline Makefile]: On CentOS 5 where we build the linux tarball binaries, the cert.pem file comes from the openssl package http://mirror.centos.org/centos/5/os/x86_64/CentOS/openssl-0.9.8e-27.el5_10.4.x86_64.rpm

Comment (wildart) [inline Makefile]: case dismissed

**Comment by tkelman** [comment]
does need mentioning in license.md, but let's see if this fixes nightly travis

**Comment by josefsachsconning** [comment]
On Ubuntu, make now gives me
```
cat: /usr/lib/ssl/cert.pem: No such file or directory
cat: /usr/lib/ssl/cert.pem: No such file or directory
cat: /usr/lib/ssl/cert.pem: No such file or directory
```
I assume that's harmless, but maybe cat's stderr should be redirected to /dev/null?

**Comment by tkelman** [comment]
I think this may count as a [useless use of `cat`](http://porkmail.org/era/unix/award.html), we should probably just test for the existence of the file rather than shelling out to `cat` and string-comparing the output to see if it's empty.

**Comment by StefanKarpinski** [comment]
I'm a little skeptical that cert files can even be copyrighted. They seems like data rather than code.

**Comment by staticfloat** [comment]
@tkelman What's a better way of testing for the existence of the file?

**Comment by tkelman** [comment]
we have a few other places where we shell out to `if [ -e  ]; then echo exists; fi` that could probably work for this

**Pull request merged**